thomasbryk/4806_Project
# Rest Controller tests

Adding tests for REST controller

Testing the following endpoints:

- /api/newBookstoreOwner
    - create a BookstoreOwner
- /api/newBookstore
    - create a Bookstore for the BookstoreOwner created in the previous test 
- /api/newBook
    - create a Book for the Bookstore created in the previous test
- /api/newCustomer
    - create a Customer
- /api/addBookToCustomersShoppingCart
    - add Book to Customer's ShoppingCart (from entities created earlier
- /api/newSale
    - Create a sale for the Book in the Customer's ShoppingCart
- /api/getBookstoreOwner
- /api/getBookstoreOwners
- /api/getBookstores
- /api/getBookstoresByBookstoreOwner
- /api/getBooks
- /api/getBook
- /api/getBooksByBookstore
- /api/getBooksSoldByBookstore

The tests that create new entities validate the end point by querying the corresponding repository to validate the entity was created. The getter endpoints are tested by comparing the endpoint's response to the expected json output.
